### PR TITLE
Is valid address

### PR DESCRIPTION
**Pull request by whirm**
Fixes Tribler issue #1130

### CODE REVIEW

**Comment by NielsZeilemaker** [comment]
@whirm nice one

File: candidate.py [hunk 1]
```diff
@@ -36,7 +27,7 @@ def is_address(address):
 class Candidate(object):
 
     def __init__(self, sock_addr, tunnel):
-        assert is_address(sock_addr), sock_addr
+        assert self.is_valid_address(sock_addr), sock_addr
```
Comment (LipuFei) [inline candidate.py:30]: Can be changed to ·is_valid_address()· and the `def is_valid_address(self, address):` code on line 41 can go away.

File: endpoint.py [hunk 2]
```diff
@@ -11,6 +11,7 @@
 from twisted.internet import reactor
 
 from .candidate import Candidate
+from util import is_valid_address
```
Comment (LipuFei) [inline endpoint.py]: Doesn't seem to have effect...

File: candidate.py (hunk 1 above)
Comment (whirm) [inline candidate.py:30]: No it can't, we need to override it in the LoopbackCandidate

File: endpoint.py (hunk 2 above)
Comment (whirm) [inline endpoint.py]: yeah, it's a leftover from some assertions I added while debugging.

**Comment by whirm** [comment]
retest this please

**Comment by whirm** [comment]
@LipuFei ready

**Comment by NielsZeilemaker** [comment]
@whirm community.py still uses dispersy.is_valid_address can you fix those?

**Comment by whirm** [comment]
Sure: https://github.com/Tribler/dispersy/pull/397

**Pull request merged**